provision.sh

```diff
@@ -7,6 +7,13 @@ set -x
 # Install tools
 sudo mv /tmp/tools/* /usr/local/bin
 
+# Stop and disable dnf-automatic service
+sudo systemctl disable dnf-automatic
+sudo systemctl stop dnf-automatic
+
+# Remove the dnf-automatic package
+sudo dnf remove -y dnf-automatic
+
```

Comment:
Either remove the package, or stop/disable the systemd units, you don't need to do both. Uninstalling the package will take care of the unit files as far as I can tell:
```
%post automatic
%systemd_post dnf-automatic.timer
%systemd_post dnf-automatic-notifyonly.timer
%systemd_post dnf-automatic-download.timer
%systemd_post dnf-automatic-install.timer

%preun automatic
%systemd_preun dnf-automatic.timer
%systemd_preun dnf-automatic-notifyonly.timer
%systemd_preun dnf-automatic-download.timer
%systemd_preun dnf-automatic-install.timer

%postun automatic
%systemd_postun_with_restart dnf-automatic.timer
%systemd_postun_with_restart dnf-automatic-notifyonly.timer
%systemd_postun_with_restart dnf-automatic-download.timer
%systemd_postun_with_restart dnf-automatic-install.timer
```